package com.company;

import java.util.Scanner;

public class Exception {

    Scanner sc=new Scanner(System.in);
    public void show() {
        int a=sc.nextInt();
        int[] n=new int[a];
        int count=0;
        for (int i = 0; i <a; i++) {
            n[i] = sc.nextInt();
        }
        try {
            for (int i = 0; i < a; i++)
            {
                for (int j = i; j < a-1; j++)
                {
                    System.out.print(n[i] + " " + n[j + 1] + "\n");
                    count++;
                }
            }
            System.out.println(count);
        }catch(ArrayIndexOutOfBoundsException e){
            System.out.println(e.getMessage());
        }
    }
    public static void main(String args[])
    {
        Exception obj=new Exception();
        obj.show();
    }

}
